### PR TITLE
Make profiling lua script write out meta data

### PR DESCRIPTION
**Pull request by rgba16f**
Additionally re-organize the benchmark upload python to make only having cpu data or gpu data acceptable. It's now only an error if there is no data or either kind.

### CODE REVIEW

File: Gems/AtomLyIntegration/CommonFeatures/Assets/Scripts/Profiling/OutputProfileData.lua [hunk 1]
```diff
@@ -100,11 +97,17 @@ function OutputProfileData:OnActivate()
         -- register for ebus callbacks
         self.tickHandler = TickBus.Connect(self)
         self.profileCaptureNotificationHandler = ProfilingCaptureNotificationBus.Connect(self)
-        self.active = true
+
+        -- output test metadata
+        ProfilingCaptureRequestBus.Broadcast.CaptureBenchmarkMetadata(self.profileName, tostring(self.outputFolder) .. "/benchmark_metadata.json")
```
Comment (smurly) [inline Gems/AtomLyIntegration/CommonFeatures/Assets/Scripts/Profiling/OutputProfileData.lua:102]: this line is too long for me to see all of it, can we reformat with line continuation?

**Review by amzn-sj — APPROVED** [review]

File: Gems/AtomLyIntegration/CommonFeatures/Assets/Scripts/Profiling/OutputProfileData.lua (hunk 1 above)
Comment (rgba16f) [reply Gems/AtomLyIntegration/CommonFeatures/Assets/Scripts/Profiling/OutputProfileData.lua:102]: The line wraps for me?

**Review by smurly — APPROVED** [review]
my lua is a bit rusty, but this looks OK to me.

**Review by jromnoa — APPROVED** [review]

**Comment by jromnoa** [comment]
The `log_monitor.py` code is not needed, it's already merged from your previous PR (there's a merge conflict)

**Review by synicalsyntax — APPROVED** [review]

**Pull request merged**